plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -2521,6 +2521,42 @@ void read_only::compute_transaction(const compute_transaction_params& params, ne
         chain_plugin::handle_bad_alloc();
     } CATCH_AND_CALL(next);
 }
+
+void read_only::send_read_only_transaction(const send_read_only_transaction_params& params, next_function<send_read_only_transaction_results> next) const {
+   try {
+      auto pretty_input = std::make_shared<packed_transaction>();
+      auto resolver = make_resolver(db, abi_serializer::create_yield_function( abi_serializer_max_time ));
+      try {
+         abi_serializer::from_variant(params.transaction, *pretty_input, resolver, abi_serializer::create_yield_function( abi_serializer_max_time ));
+      } EOS_RETHROW_EXCEPTIONS(chain::packed_transaction_type_exception, "Invalid packed transaction")
+
+      app().get_method<incoming::methods::transaction_async>()(pretty_input, false /* api_trx */, transaction_metadata::trx_type::read_only, true /* return_failure_trace */,
```

Comment:
Both here and compute_transaction, `api_trx` should be `true`. It is only used (in `main`) to determine which subjective billing configuration to honor. Subjective billing is not applicable to read_only and compute_transaction, but these are both technically api trxs (and not p2p transactions which is what `false` means).

Reply:
Thanks for the explanation.